### ISSUE TITLE
prep template add failure when there are extra white spaced columns

### ISSUE THREAD

**Issue by antgonza** [biocore/qiita#757]
If the prep template has blank columns at the end, the script will fail without a good explanation.

The traceback is:
```
ERROR:
nan
TRACE:
Traceback (most recent call last):
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/web.py", line 1144, in _when_complete
if result.result() is not None:
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/concurrent.py", line 129, in result
raise_exc_info(self.__exc_info)
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/gen.py", line 221, in wrapper
runner.run()
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/gen.py", line 505, in run
yielded = self.gen.throw(*exc_info)
File "/Users/antoniog/svn_programs/qiita/qiita_pet/handlers/study_handlers/description_handlers.py", line 593, in post
study, user)
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/gen.py", line 533, in run
self.yield_point.start(self)
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tornado/gen.py", line 371, in start
self.func(*self.args, **self.kwargs)
File "/Users/antoniog/svn_programs/qiita/qiita_pet/handlers/study_handlers/description_handlers.py", line 284, in add_prep_template
investigation_type)
File "/Users/antoniog/svn_programs/qiita/qiita_pet/handlers/study_handlers/description_handlers.py", line 479, in remove_add_prep_template
investigation_type=investigation_type).id
File "/Users/antoniog/svn_programs/qiita/qiita_db/metadata_template.py", line 1316, in create
missing = cls._check_special_columns(md_template, data_type_str)
File "/Users/antoniog/svn_programs/qiita/qiita_db/metadata_template.py", line 674, in _check_special_columns
[handler[i] for i in md_template[value]],
KeyError: nan
```

**Comment by ElDeveloper** [comment]
Do you have a template you can send me to see if I can reproduce this issue? I'm afraid, this might be a consequence due to the way that we load templates using pandas. May be related to #740 (also taking a look at it right now).

**Comment by ElDeveloper** [comment]
Looked into it and the problem was an empty row at the end of the file, not an empty column:

```python
In [18]: bad = load_template_to_dataframe('../../original.txt')

In [19]: len(bad)
Out[19]: 119

In [22]: good = load_template_to_dataframe('../../corrected.txt')

In [23]: len(good)
Out[23]: 118
```

This problem will be fixed by #759 (I'm amending the commit message so this issue is also closed onces 759 is merged).
